File: 1685-sum-of-absolute-differences-in-a-sorted-array/1685-sum-of-absolute-differences-in-a-sorted-array.cpp
class Solution {
public:
    vector<int> getSumAbsoluteDifferences(vector<int>& nums) 
    {
        // two pointer approach
        // keep track of pre sum and post sum
        vector<int> res(nums.size(), 0);
        int pre{0}, pos = accumulate(nums.begin(), nums.end(), 0);
        
        for(int i=0; i<nums.size(); ++i)
        {
            pos -= nums[i];
            res[i] = pos - (nums[i]*(nums.size()-i-1)) + (nums[i]*i - pre);
            pre += nums[i];
        }
        
        return res;
        
    }
};

File: 1685-sum-of-absolute-differences-in-a-sorted-array/1685-sum-of-absolute-differences-in-a-sorted-array.py
class Solution:
    def getSumAbsoluteDifferences(self, nums: List[int]) -> List[int]:
        # two pointer approach
        # keep track of preSum and postSum
        # apply formula to get sum of abs difference at each i
        pre = 0
        pos = sum(nums)
        n = len(nums)
        res = [0]*n
        for i in range(0, n):
            pos -= nums[i]
            res[i] = pos - nums[i]*(n-i-1) + ((nums[i])*i - pre)
            pre += nums[i]
            
        return res